ui/apps/platform/src/Containers/NetworkGraph/flows/AdvancedFlowsFilter.tsx

```diff
@@ -0,0 +1,149 @@
+        } else {
+            filters.ports.push(selection);
+        }
+    });
+    return filters;
+}
+
+function AdvancedFlowsFilter({
+    filters,
+    setFilters,
+}: AdvancedFlowsFilterProps): React.ReactElement {
+    // derived state
+    const selections = filtersToSelections(filters);
+
+    // component state
+    const [isFilterDropdownOpen, setIsFilterDropdownOpen] = React.useState(false);
+    const {
+        isOpen: isPortsSelectOpen,
+        onToggle: onTogglePortsSelect,
+        onSelect: onSelectPorts,
+    } = useMultiSelect(handlePortsSelect, filters.ports, false);
+
+    // setters
+    const onFilterDropdownToggle = (isOpen: boolean) => {
+        setIsFilterDropdownOpen(isOpen);
+    };
+    const onTrafficFilterSelect = (event, selection) => {
+        if (selections.includes(selection)) {
+            setFilters((prevFilters) => {
+                const prevSelection = filtersToSelections(prevFilters);
+                const newSelection = prevSelection.filter((item) => item !== selection);
+                const newFilters = selectionsToFilters(newSelection);
+                return newFilters;
+            });
+        } else {
+            setFilters((prevFilters) => {
+                const prevSelection = filtersToSelections(prevFilters);
+                const newSelection = [...prevSelection, selection] as string[];
+                const newFilters = selectionsToFilters(newSelection);
+                return newFilters;
+            });
+        }
+    };
+    function handlePortsSelect(selection) {
+        setFilters((prevFilters) => {
+            const newFilters = { ...prevFilters };
+            newFilters.ports = selection;
+            return newFilters;
+        });
+    }
+
+    return (
+        <Select
+            className="advanced-flows-filters-select"
+            variant={SelectVariant.checkbox}
+            onToggle={onFilterDropdownToggle}
+            onSelect={onTrafficFilterSelect}
+            selections={selections}
+            isOpen={isFilterDropdownOpen}
+            placeholderText="Advanced"
+            aria-labelledby="advanced-flows-filters-select"
+            isGrouped
+            position={SelectPosition.right}
+        >
+            <SelectGroup label="Deployment flows">
+                <SelectOption value="anomalous">Anomalous flows</SelectOption>
+                <SelectOption value="baseline">Baselined flows</SelectOption>
+            </SelectGroup>
+            <SelectGroup label="Flow directionality">
+                <SelectOption value="ingress">Ingress (inbound)</SelectOption>
+                <SelectOption value="egress">Egress (outbound)</SelectOption>
+            </SelectGroup>
+            <SelectGroup label="Protocols">
+                <SelectOption value="TCP">TCP</SelectOption>
+                <SelectOption value="UDP">UDP</SelectOption>
+            </SelectGroup>
+            <SelectGroup label="Ports">
+                <Select
```

Comment:
This is a bit of an odd UX flow. Are we planning to later "pre-seed" the list of ports, based on what's available? Otherwise, every port will be a new "createable" entry--in which case, is there a PF text-input-with-chips component that would be clearer?

@alanonthegit The UX team may have already tried different solutions and found this was the best available, but I figure it's worth asking.

Reply:
Yea, I was thinking about that after I sent out the PR (https://github.com/stackrox/stackrox/pull/3780#discussion_r1019775427). I can actually pass all unique ports as select options. I think that would help make the UX feel less odd